NitroxClient/GameLogic/Bases/BuildingHandler.cs

```diff
@@ -0,0 +1,388 @@
+            FailedOperations++;
+            Log.Debug("Couldn't find the parentBase");
+            yield break;
+        }
+        BaseDeconstructable[] deconstructableChildren = baseObject.GetComponentsInChildren<BaseDeconstructable>(true);
+        Log.Debug($"Found {deconstructableChildren.Length} BaseDeconstructable in the base");
+
+        if (deconstructableChildren.Length == 1 && deconstructableChildren[0])
+        {
+            Log.Debug("Will only deconstruct the base manually");
+            using (new PacketSuppressor<BaseDeconstructed>())
+            using (new PacketSuppressor<PieceDeconstructed>())
+            {
+                deconstructableChildren[0].Deconstruct();
+            }
+            BasesCooldown[baseDeconstructed.FormerBaseId] = DateTimeOffset.Now;
+            yield break;
+        }
+        EnsureTracker(baseDeconstructed.FormerBaseId).FailedOperations++;
+    }
+
+    public IEnumerator DeconstructPiece(PieceDeconstructed pieceDeconstructed)
+    {
+        if (!NitroxEntity.TryGetComponentFrom(pieceDeconstructed.BaseId, out Base @base))
+        {
+            FailedOperations++;
+            Log.Debug("Couldn't find the parentBase");
+            yield break;
+        }
+
+        OperationTracker tracker = EnsureTracker(pieceDeconstructed.BaseId);
+
+        BuildPieceIdentifier pieceIdentifier = pieceDeconstructed.BuildPieceIdentifier;
+        Transform cellObject = @base.GetCellObject(pieceIdentifier.BaseCell.ToUnity());
+        if (!cellObject)
+        {
+            Log.Debug($"Couldn't find cell object {pieceIdentifier.BaseCell} when destructing piece {pieceDeconstructed}");
+            yield break;
+        }
+        BaseDeconstructable[] deconstructableChildren = cellObject.GetComponentsInChildren<BaseDeconstructable>(true);
+        // TODO: Fix this
+        foreach (BaseDeconstructable baseDeconstructable in deconstructableChildren)
+        {
+            if (!BuildUtils.TryGetIdentifier(baseDeconstructable, out BuildPieceIdentifier identifier) || !identifier.Equals(pieceIdentifier))
+            {
+                continue;
+            }
+            Log.Debug($"[{pieceDeconstructed.OperationId}] Found a BaseDeconstructable {baseDeconstructable.name}, will now deconstruct it manually");
+            using (PacketSuppressor<BaseDeconstructed>.Suppress())
+            using (PacketSuppressor<PieceDeconstructed>.Suppress())
+            using (PacketSuppressor<WaterParkDeconstructed>.Suppress())
+            {
+                Temp.Fill(pieceDeconstructed);
+                baseDeconstructable.Deconstruct();
+                Temp.Reset();
+            }
+            tracker.RegisterOperation(pieceDeconstructed.OperationId);
+            BasesCooldown[pieceDeconstructed.BaseId] = DateTimeOffset.Now;
+            yield break;
+        }
+        Log.Error("Couldn't find BaseDeconstructable to be destructed");
+        tracker.FailedOperations++;
+    }
+
+    public void AskForResync()
```

Comment:
If I created a new class for that, it should also be a single instance containg all fo the resync logic. I would make it a MB attached to the Multiplayer GO as well. It would still need to interact with BuildingHandler.
What do you think ?